### PR TITLE
Remove uprobes-based manual instrumentation

### PR DESCRIPTION
**Pull request by dpallotti**
In particular, all the code related to "Orbit function types" can be removed.

See notes in the bug why we believe this change is not problematic.

Bug: http://b/194288616

Test:
- Open a capture taken with main on `OrbitTestLegacyApi`: As expected, many
  things regarding manual instrumentation are now wrong, but the capture can
  still be opened and the rest is correct.
- Open a capture taken with main on `OrbitTest`, verify no information was lost.
- Take a capture on `OrbitTest`, verify manual instrumentation.
- Take a capture on `OrbitTest` also with dynamic instrumentation and try
  various filters in the capture view.

### CODE REVIEW

File: src/MetricsUploader/CaptureMetricTest.cpp [hunk 1]
```diff
@@ -49,19 +44,14 @@ bool HasSameCaptureStartData(const OrbitCaptureData& capture_data,
...
-             start_data.number_of_manual_tracked_values &&
          capture_data.thread_states() == start_data.thread_states &&
          capture_data.memory_information_sampling_period_ms() ==
-             start_data.memory_information_sampling_period_ms;
+             start_data.memory_information_sampling_period_ms &&
+         capture_data.lib_orbit_vulkan_layer() == start_data.lib_orbit_vulkan_layer &&
+         capture_data.local_marker_depth_per_command_buffer() ==
+             start_data.local_marker_depth_per_command_buffer &&
+         capture_data.max_local_marker_depth_per_command_buffer() ==
+             start_data.max_local_marker_depth_per_command_buffer;
```
Comment (dimitry-) [inline src/MetricsUploader/CaptureMetricTest.cpp:54]: I wonder how we can make it more obvious. so that these are not missed in the future.

**Review by dimitry- — APPROVED** [review]

**Review by florian-kuebler — APPROVED** [review]
This is great! Thanks a lot!

**Review by pierricgimmig — APPROVED** [review]
LGTM, thanks for taking care of this.

File: src/MetricsUploader/CaptureMetricTest.cpp (hunk 1 above)
Comment (dpallotti) [reply src/MetricsUploader/CaptureMetricTest.cpp:54]: Hmm, no idea, especially since one is a proto and one is a regular `struct`.

**Pull request merged**